homeassistant/components/edl21/config_flow.py

```diff
@@ -0,0 +1,23 @@
+from homeassistant import config_entries, core
+import voluptuous as vol
+from .sensor import DOMAIN, CONF_SERIAL_PORT
+from homeassistant.const import CONF_NAME
+import homeassistant.helpers.config_validation as cv
+
+
+EDL_SCHEMA = vol.Schema(
+    {vol.Required(CONF_SERIAL_PORT): cv.string, vol.Optional(CONF_NAME): cv.string}
```

Comment:
Instead of asking for a string as a serial port, we should make it a dropdown, so a user can just pick a serial port from a list. The configuration flow of ZHA might be a good example for this one.

Reply:
It is also possible to use a ser2net socket (which you can't pick from a list) instead of a serial port.

Reply:
In that case, I  would recommend creating a multi-step configuration flow so both those cases can be handled.

Reply:
I understand. So lets keep this PR small and simple and extend/improve this in an other PR?

Reply:
now I added listing of serial ports as radio buttons.